src/Editor.js

```diff
@@ -1332,99 +1338,106 @@ const Editor = () => {
                 config[
                   doc.layers.find(({ id }) => id === [...view.selection][0])
                     .component
-                ].options?.map(({ key, input, label }) => {
+                ].options?.map(({ key, input, label }, index) => {
+                  const layer = doc.layers.find(({ id }) => id === [...view.selection][0])
+                  const error = config[layer.component].validate(layer.options, key)
```

Comment:
We shouldn't need to pass the `key` into the `validate()` function because the data it returns has a `key` property that we can use to get the relevant errors here (the `.?` is an [optional chaining operator](https://github.com/tc39/proposal-optional-chaining) to handle the case where `validate()` returns `null` and the `.find(() => true)` call is used to get the first item off the array):

```suggestion
                  const error = config[layer.component].validate(layer.options)?.filter(error => error.key === key).find(() => true);
```

That means `validate()` functions that check multiple options/rules need to return an array of all errors instead of a single error.